src/Simplify_EQ.cpp

```diff
@@ -39,6 +42,12 @@ Expr Simplify::visit(const EQ *op, ExprInfo *bounds) {
     Expr delta = mutate(op->a - op->b, &delta_bounds);
     const int lanes = op->type.lanes();
 
+    // If the bounds are 0, then it's just x == x
```

Comment:
I don't think this actually covers floating point x == x, as the bounds won't be defined. I think the missing rewrite (in addition to the one above), is actually 0 == 0

Reply:
Though written like that it's obviously not going to be a matcher. Maybe we should just say if (is_zero(delta)) for the condition here.

Reply:
Ah right, I forgot bounds won't be defined for floating point.